### PR TITLE
Feat: Added Result<T, E> support

### LINKED ISSUE #171: Support the Result<V, E> type
I suggest that a default implementation is addded for the Result<V, E> type of the standard library

### PR DESCRIPTION
**Pull request by KronsyC**
Resolves #171

### CODE REVIEW

File: ts-rs/src/lib.rs [hunk 1]
```diff
@@ -459,6 +459,23 @@ impl<T: TS> TS for Option<T> {
...
+        unreachable!();
+    }
+    fn transparent() -> bool {
+        return true;
+    }
+    fn dependencies() -> Vec<Dependency>
+    where
+        Self: 'static {
+        [Dependency::from_ty::<T>()].into_iter().flatten().collect()
+    }
```
Comment (axos88) [inline ts-rs/src/lib.rs:473]: I think the dependency on E is missing

File: ts-rs/src/lib.rs [hunk 2]
```diff
@@ -459,6 +459,23 @@ impl<T: TS> TS for Option<T> {
...
+    fn inline() -> String {
+        format!("{{ ok : {} }} | {{ err : {} }}", T::inline(), E::inline())
+    }
```
Comment (axos88) [inline ts-rs/src/lib.rs:476]: This should be capital cased though,  shouldn't it?

**Comment by KronsyC** [comment]
I have addressed both of your issues in the most recent commit

**Comment by NyxCode** [comment]
Thanks @KronsyC for the PR, and thanks @axos88 for taking a look at it. Really appreciate it!

**Pull request merged**